string mode
Float64 desired_distance
---
bool success
